Fauna-RPG/Entity/Components/BuffComponent.gd

```diff
@@ -0,0 +1,33 @@
+extends EntityComponent
+class_name BuffComponent
+## Applies buffs to the entity and holds a list of current buffs.
+
+
+var current_buffs : Array[BuffBase]
```

Comment:
Instead of an `Array`, maybe a `Dictionary` would make more sense, especially if the same type of buff can stack (strength for example)

Reply:
You could have something like that:
`buff_dict: Dictionnary[Buff] = {Buff_name: Buff : (Buff_value: float, Buff_duration: int)}`

Reply:
Now that I think about it, I think we can keep the Array, but just add to the duration directly. Like so:

```
# see if the buff was already applied. If so, add to the duration instead of applying
var buff_index: int = current_buffs.find(buff_copy)
if buff_index != -1:
     current_buffs[buff_index].buff_turn_duration += buff_copy.buff_turn_duration
```

Reply:
I generally prefer dict because of the key/value structure which i find easier than to use a find function every time we need to change something. But both the array and the dict would work for this.

Reply:
I'll push the Array version for now and we can change to a dict if it becomes too cumbersome